### PR TITLE
Poll/Block: Ensure proper input/output handling

### PR DESCRIPTION
**Pull request by mjansenDatabay**
See: https://mantis.ilias.de/view.php?id=41229

If approved, this MUST be picked to `release_9` and `trunk`.

### CODE REVIEW

File: Services/Block/classes/class.ilBlockGUI.php [hunk 1]
```diff
@@ -876,22 +886,22 @@ public function getHTMLNew(): string
         switch ($this->getPresentation()) {
             case self::PRES_SEC_LEG:
                 $panel = $factory->panel()->secondary()->legacy(
-                    $this->getTitle(),
+                    $this->specialCharsAsEntities($this->getTitle()),
```
Comment (mjansenDatabay) [inline Services/Block/classes/class.ilBlockGUI.php:889]: One can debate about if this (and the other five changed lines) belongs to `ilBlockGUI` or the respective UI component, but there is already some similar "title handling/string transformation" in `ilBlockGUI`.

**Comment by mjansenDatabay** [comment]
Hi @smeyer-ilias , do you have any objections here? I highly recommend merging this (or providing another solution).

**Comment by schmitz-ilias** [comment]
The changes in Poll look good to me, thanks for the PR @mjansenDatabay!

 I guess according to the new procedure regarding PRs into unmaintained code I'm also shepherding the changes in Block? In any case, those changes are slight and seem to work fine, so I'll merge and pull the PR up to 9 and trunk (with the necessary changes due to refactorings in Poll in 9).

**Comment by mjansenDatabay** [comment]
@schmitz-ilias Thanks a lot! You are right, it's kind of shepherding the `ilBlockGUI` changes. As I (tried to) explain(ed) in the JourFixe, this does not give you any new obligations. It's just a "Review" (and maybe a short "Shakedown Test").

**Pull request merged**